service/controller/appcatalog/resource/appcatalogentry/resource.go

```diff
@@ -116,3 +116,41 @@ func (r *Resource) getIndex(ctx context.Context, storageURL string) (index, erro
 
 	return i, nil
 }
+
+func (r *Resource) getMetadata(ctx context.Context, storageURL, name, version string) ([]byte, error) {
+	mainURL := fmt.Sprintf("%s/%s-%s.tgz-meta/main.yaml", strings.TrimRight(storageURL, "/"), name, version)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting main.yaml from %#q", mainURL))
+
+	// We use https in catalog URLs so we can disable the linter in this case.
+	resp, err := http.Get(mainURL) // #nosec
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no main.yaml generated in %#q", mainURL))
```

Comment:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no main.yaml generated at %#q", mainURL))
```

Minor but `in` reads a bit strange for me. The file is at the URL if it exists.

Reply:
it makes sense, I'm going to change that. Thanks for the suggestion.